### PR TITLE
feat(cosmos send): add memo support

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

This adds support for memos in Cosmos (SDK) send modals

- display a memo field
- support a maximum memo length of 256 chars: there is no real spec/limitation in Cosmos SDK for this - it could be any go uint64 so technically up to 18,446,744,073,709,551,615 , but the auth module uses 256 as an example for `MaxMemoCharacters` https://github.com/cosmos/cosmos-sdk/blob/2627a96a496e3c47d97b5c34720a7359600b70ed/x/auth/spec/06_params.md
- shows the amount of remaining characters and doesn't let the user continue with the flow if the limit is exceeded
- on the next modal, the memo is displayed when confirming the transaction

## Notice

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

#651

## Risk

N/A

## Testing

- enable the Cosmos flags
- go to the ATOM (Cosmos) asset
- try to send some ATOM and add a memo

- memo should be accepted and it should be possible to continue the flow if it's less or equal to 256 chars
- memo should not be accepted and it should be impossible to continue the flow if 256 chars are exceeded - continue button is disabled, and in errored state withand the characters counter turns red
- any string should work as long as it is less or equal to 256 chars - tested using keplr and checking cosmos-sdk docs and code, there is no check except than this one 

## Screenshots (if applicable)

image

image
﻿
image

image

image

### CODE REVIEW

File: src/plugins/cosmos/components/modals/Send/views/Confirm.tsx [hunk 1]
```diff
@@ -0,0 +1,162 @@
+import { ArrowBackIcon } from '@chakra-ui/icons'
```
Comment (gomesalexandre) [inline src/plugins/cosmos/components/modals/Send/views/Confirm.tsx:1]: This module is a copy of the existing `<Confirm />` with the addition of:
https://github.com/shapeshift/web/pull/1238/files#diff-1d0d4750995979e616dce4e87f33fa932399a6e7f908670a3981af2a4a18a8bfR99-R108

The reason for the duplication is that it consumes the Cosmos `SendInput` which is different for the existing one, and trying to use both `ensName` and `memo` on the existing `<Confirm />` would make us fight with types and is also blending too much chain-specific stuff in a general-purpose component, while chain-specific logic should live in plugins.

File: src/plugins/cosmos/components/modals/Send/views/Details.tsx [hunk 2]
```diff
@@ -0,0 +1,259 @@
+import { ArrowBackIcon } from '@chakra-ui/icons'
```
Comment (gomesalexandre) [inline src/plugins/cosmos/components/modals/Send/views/Details.tsx:1]: This module is a duplication of the existing `<Details />` with the addition of these JSX nodes:
https://github.com/shapeshift/web/pull/1238/files#diff-32a7b208eee8c7e78e369610d1f12492c5538b5dbaa8ff9f7c0540de4ad2a6e5R185-R232

The reason for the duplication is that it consumes the Cosmos `SendInput` which is different for the existing one, and trying to use both `ensName` and `memo` on the existing `<Confirm />` would make us fight with types and is also blending too much chain-specific stuff in a general-purpose component, while chain-specific logic should live in plugins.

**Review by DaoDev44 — APPROVED** [review]
Tested locally and everything works as intended. LGTM

File: package.json [hunk 3]
```diff
@@ -25,9 +25,9 @@
     "@shapeshiftoss/hdwallet-native-vault": "^1.19.0",
     "@shapeshiftoss/hdwallet-portis": "^1.19.0",
     "@shapeshiftoss/investor-yearn": "^1.4.3",
+    "@shapeshiftoss/types": "^2.3.0",
     "@shapeshiftoss/market-service": "^2.1.0",
     "@shapeshiftoss/swapper": "^2.0.2",
```
Comment (mrnerdhair) [inline package.json:30]:
```suggestion
    "@shapeshiftoss/market-service": "^2.1.0",
    "@shapeshiftoss/swapper": "^2.0.2",
    "@shapeshiftoss/types": "^2.3.0",
```

**Review by mrnerdhair — APPROVED** [review]

**Review by 0xdef1cafe — APPROVED** [review]

**Pull request merged**